pages/collective-page.tsx

```diff
@@ -0,0 +1,192 @@
+import { getSSRQueryHelpers } from '../lib/apollo-client';
+import { getCollectivePageMetadata } from '../lib/collective';
+import { OPENCOLLECTIVE_FOUNDATION_ID } from '../lib/constants/collectives';
+import { generateNotFoundError } from '../lib/errors';
+import useLoggedInUser from '../lib/hooks/useLoggedInUser';
+import { PREVIEW_FEATURE_KEYS } from '../lib/preview-features';
+import { addParentToURLIfMissing, getCollectivePageCanonicalURL } from '../lib/url-helpers';
+import { getRequestIntl } from '@/lib/i18n/request';
+
+import CollectivePageContent from '../components/collective-page';
+import CollectiveNotificationBar from '../components/collective-page/CollectiveNotificationBar';
+import CollectiveThemeProvider from '../components/CollectiveThemeProvider';
+import { CrowdfundingPreviewBanner } from '../components/crowdfunding-redesign/CrowdfundingPreviewBanner';
+import ErrorPage from '../components/ErrorPage';
+import Loading from '../components/Loading';
+import Page from '../components/Page';
+import { preloadCollectivePageGraphqlQueries } from '@/components/collective-page/graphql/preload';
+import { collectivePageQuery } from '@/components/collective-page/graphql/queries';
+import { MAX_CONTRIBUTORS_PER_CONTRIBUTE_CARD } from '@/components/contribute-cards/constants';
+import GuestUserProfile from '@/components/GuestUserProfile';
+import IncognitoUserCollective from '@/components/IncognitoUserCollective';
+import OnboardingModal from '@/components/onboarding-modal/OnboardingModal';
+
+import Custom404 from './404';
+
+const GlobalStyles = createGlobalStyle`
+  section {
+    margin: 0;
+  }
+`;
+
+const getVariablesFromContext = ({ query }: Context) => {
+  return {
+    slug: query.slug,
+    nbContributorsPerContributeCard: MAX_CONTRIBUTORS_PER_CONTRIBUTE_CARD,
+  };
+};
+
+const getPropsFromContext = (context: Context) => {
+  const {
+    query: { slug, status, step, mode, action },
+  } = context;
+
+  return { slug, status, step, mode, action };
+};
+
+const collectivePageQueryHelper = getSSRQueryHelpers<
+  ReturnType<typeof getVariablesFromContext>,
+  ReturnType<typeof getPropsFromContext>
+>({
+  query: collectivePageQuery,
+  getVariablesFromContext,
+  getPropsFromContext,
+  preload: (client, result) => preloadCollectivePageGraphqlQueries(client, result?.data?.Collective),
+});
+
+// next.js export
+// ts-unused-exports:disable-next-line
+export const getServerSideProps = async (context: Context) => {
+  const result = await collectivePageQueryHelper.getServerSideProps(context);
+  const props = result['props'];
+  const notFound = collectivePageQueryHelper.checkResultContainsNonNullResult(props, 'Collective(');
+  if (context.res && context.req) {
+    const { locale } = getRequestIntl(context.req);
+    if (locale === 'en') {
+      context.res.setHeader('Cache-Control', 'public, s-maxage=300');
+    }
+  }
+
+  if (notFound) {
+    return {
+      notFound: true,
+    };
+  }
```

Comment:
I don't think it is right to assume `null` = the collective doesn't exist. You could end up in the same situation with a 500 Internal server error, and collective admins would freak out thinking that their collective is deleted when we get a server crash. We should not obfuscate it.